Marked files for translation

awaisalee opened the pull request:
Marked files for translation:

- packages/desktop-client/src/components/UpdateNotification.tsx
- packages/desktop-client/src/components/sidebar/PrimaryButtons.tsx
- packages/desktop-client/src/components/transactions/TransactionsTable.jsx
- packages/loot-core/src/platform/server/fs/index.electron.ts
- packages/loot-core/src/platform/server/fs/index.web.ts
- packages/loot-core/src/platform/server/fs/shared.ts
- packages/loot-core/src/server/accounts/qif2json.ts
- packages/loot-core/src/server/app.ts
- packages/loot-core/src/server/aql/exec.ts
- packages/loot-core/src/server/aql/schema/executors.ts
- packages/desktop-client/src/components/ManageRules.tsx
- packages/desktop-client/src/components/ManageRulesPage.tsx
- packages/desktop-client/src/components/Notes.tsx
- packages/desktop-client/src/components/NotesButton.tsx
- packages/desktop-client/src/components/ThemeSelector.tsx
- packages/desktop-client/src/components/Titlebar.tsx
- packages/desktop-client/src/components/mobile/MobileBackButton.tsx
- packages/desktop-client/src/components/mobile/accounts/Accounts.jsx
- packages/desktop-client/src/components/mobile/budget/BudgetTable.jsx
- packages/desktop-client/src/components/mobile/transactions/AddTransactionButton.tsx
- packages/desktop-client/src/components/mobile/transactions/Transaction.jsx
- packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx
- packages/desktop-client/src/components/mobile/transactions/TransactionListWithBalances.jsx
- packages/desktop-client/src/components/modals/AccountAutocompleteModal.tsx
- packages/desktop-client/src/components/modals/AccountMenuModal.tsx
- packages/desktop-client/src/components/modals/BudgetListModal.tsx
- packages/desktop-client/src/components/modals/CategoryAutocompleteModal.tsx
- packages/desktop-client/src/components/modals/CategoryGroupMenuModal.tsx
- packages/desktop-client/src/components/modals/CategoryMenuModal.tsx
- packages/desktop-client/src/components/modals/CloseAccountModal.tsx
- packages/desktop-client/src/components/modals/ConfirmCategoryDeleteModal.tsx
- packages/desktop-client/src/components/modals/ConfirmTransactionDeleteModal.tsx
- packages/desktop-client/src/components/modals/ConfirmTransactionEditModal.tsx
- packages/desktop-client/src/components/modals/ConfirmUnlinkAccountModal.tsx
- packages/desktop-client/src/components/modals/CreateAccountModal.tsx
- packages/desktop-client/src/components/modals/CreateEncryptionKeyModal.tsx
- packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx
- packages/desktop-client/src/components/modals/EditFieldModal.jsx
- packages/desktop-client/src/components/modals/EditRuleModal.jsx
- packages/desktop-client/src/components/modals/FixEncryptionKeyModal.tsx
- packages/desktop-client/src/components/modals/GoCardlessInitialiseModal.tsx
- packages/desktop-client/src/components/modals/HoldBufferModal.tsx
- packages/desktop-client/src/components/modals/ImportTransactionsModal/ImportTransactionsModal.jsx
- packages/desktop-client/src/components/modals/KeyboardShortcutModal.tsx
- packages/desktop-client/src/components/modals/ManageRulesModal.tsx
- packages/desktop-client/src/components/modals/MergeUnusedPayeesModal.jsx
- packages/desktop-client/src/components/modals/NotesModal.tsx
- packages/desktop-client/src/components/modals/PayeeAutocompleteModal.tsx
- packages/desktop-client/src/components/modals/ScheduledTransactionMenuModal.tsx
- packages/desktop-client/src/components/modals/SelectLinkedAccountsModal.jsx
- packages/desktop-client/src/components/modals/SimpleFinInitialiseModal.tsx
- packages/desktop-client/src/components/payees/PayeeMenu.tsx
- packages/desktop-client/src/components/reports/CategorySelector.tsx
- packages/desktop-client/src/components/reports/ReportSidebar.tsx
- packages/desktop-client/src/components/reports/ReportSummary.tsx
- packages/desktop-client/src/components/reports/ReportTopbar.tsx
- packages/desktop-client/src/components/reports/SaveReport.tsx
- packages/desktop-client/src/components/reports/SaveReportDelete.tsx
- packages/desktop-client/src/components/reports/SaveReportName.tsx
- packages/desktop-client/src/components/reports/reports/CustomReport.tsx
- packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx
- packages/desktop-client/src/components/reports/reports/GetCardData.tsx
- packages/desktop-client/src/components/rules/ActionExpression.tsx
- packages/desktop-client/src/components/rules/RuleRow.tsx
- packages/desktop-client/src/components/rules/RulesHeader.tsx
- packages/desktop-client/src/components/rules/Value.tsx
- packages/desktop-client/src/components/settings/BudgetTypeSettings.tsx
- packages/desktop-client/src/components/settings/Encryption.tsx
- packages/desktop-client/src/components/settings/Export.tsx
- packages/desktop-client/src/components/settings/FixSplits.tsx
- packages/desktop-client/src/components/settings/Format.tsx
- packages/desktop-client/src/components/settings/Reset.tsx
- packages/desktop-client/src/components/settings/Themes.tsx
- packages/desktop-client/src/components/settings/UI.tsx
- packages/desktop-client/src/components/settings/index.tsx
- packages/desktop-client/src/components/transactions/SimpleTransactionsTable.jsx
- packages/desktop-client/src/components/util/AmountInput.tsx
- packages/desktop-client/src/global-events.ts

MikesGlitch commented on packages/desktop-client/src/components/settings/Export.tsx:
```diff
@@ -27,14 +28,14 @@ export function ExportBudget() {
       setError(response.error);
       setIsLoading(false);
-      console.log('Export error code:', response.error);
+      console.log(t('Export error code:'), response.error);
```
```suggestion
      console.log('Export error code:', response.error);
```

Shouldn't need to translate console logs

MikesGlitch commented on packages/loot-core/src/platform/server/fs/index.electron.ts:
```diff
@@ -146,7 +155,9 @@ export const writeFile: T.WriteFile = async (filepath, contents) => {
     return undefined;
   } catch (err) {
-    console.error(`Unable to recover from file lock on file ${filepath}`);
+    console.error(
```
No need to translate error logs

MikesGlitch commented on packages/loot-core/src/platform/server/fs/index.web.ts:
```diff
@@ -77,15 +78,15 @@ async function _readFile(filepath, opts?: { encoding?: string }) {
   ) {
     if (!_exists(filepath)) {
-      throw new Error('File does not exist: ' + filepath);
+      throw new Error(t('File does not exist:') + ' ' + filepath);
```
Same here, these aren't shown to the user so no need to translate them. Ideally they'd be english so we can read them in English when people come for support

MikesGlitch commented on packages/loot-core/src/platform/server/fs/index.web.ts:
```diff
@@ -77,15 +78,15 @@ async function _readFile(filepath, opts?: { encoding?: string }) {
 
     if (item == null) {
-      throw new Error('File does not exist: ' + filepath);
+      throw new Error(t('File does not exist:') + ' ' + filepath);
```
Here too - I wont put a comment on all of them but it would be ideal to remove them all 👍

MikesGlitch commented on packages/loot-core/src/platform/server/fs/shared.ts:
```diff
@@ -1,19 +1,21 @@
 export const getDocumentDir = () => {
   if (!documentDir) {
-    throw new Error('Document directory is not set');
+    throw new Error(t('Document directory is not set'));
```
No need to translate thrown errors (check rest of this file for the same issue)

MikesGlitch commented on packages/loot-core/src/server/accounts/qif2json.ts:
```diff
@@ -96,11 +100,10 @@ export function qif2json(qif, options: { dateFormat?: string } = {}) {
 
       default:
-        throw new Error('Unknown Detail Code: ' + line[0]);
+        throw new Error(t('Unknown Detail Code: {{code}}', { code: line[0] }));
```
No need to translate thrown errors (check rest of this file for the same issue)

MikesGlitch commented on packages/loot-core/src/server/app.ts:
```diff
@@ -27,7 +28,9 @@ class App<Handlers> {
     if (this.handlers[name] != null) {
       throw new Error(
-        'Conflicting method name, names must be globally unique: ' + name,
+        t('Conflicting method name, names must be globally unique: {name}', {
```
No need to translate thrown errors (check rest of this file for the same issue)

MikesGlitch commented on packages/loot-core/src/server/aql/exec.ts:
```diff
@@ -47,7 +49,7 @@ export async function runCompiledQuery(
     const name = param.paramName;
     if (params[name] === undefined) {
-      throw new Error(`Parameter ${name} not provided to query`);
+      throw new Error(t('Parameter {name} not provided to query', { name }));
```
No need to translate thrown errors (check rest of this file for the same issue)

MikesGlitch commented on packages/loot-core/src/server/aql/schema/executors.ts:
```diff
@@ -32,7 +34,11 @@ function execTransactions(state, query, sql, params, outputTypes) {
   if (['all', 'inline', 'none', 'grouped'].indexOf(splitType) === -1) {
-    throw new Error(`Invalid “splits” option for transactions: “${splitType}”`);
+    throw new Error(
+      t('Invalid “splits” option for transactions: “{{splitType}}”', {
```
No need to translate thrown errors (check rest of this file for the same issue)

MikesGlitch commented on packages/desktop-client/src/components/rules/ActionExpression.tsx:
```diff
@@ -70,10 +72,10 @@ function SetActionExpression({
       <Text>{friendlyOp(op)}</Text>{' '}
       <Text style={valueStyle}>{mapField(field, options)}</Text>{' '}
-      <Text>to </Text>
+      <Text>{t('to')}</Text>
```
Missing a space after _to_

MikesGlitch commented on packages/desktop-client/src/components/rules/ActionExpression.tsx:
```diff
@@ -70,10 +72,10 @@ function SetActionExpression({
       {options?.template ? (
         <>
-          <Text>template </Text>
+          <Text>{t('template')}</Text>
```
Missing a space after _template_

MikesGlitch reviewed (commented):
This looks great! I've added some minor points about translating errors. Ideally we don't want to do that - those errors are for devs and for support - they shouldn't be shown to the user so shouldn't be translated. 

E2E tests are failing as well - if you need help let us know.

awaisalee commented:
> This looks great! I've added some minor points about translating errors. Ideally we don't want to do that - those errors are for devs and for support - they shouldn't be shown to the user so shouldn't be translated.
> 
> E2E tests are failing as well - if you need help let us know.

Hi @MikesGlitch. I hope you're doing well. Thanks a lot for reviewing my PR. Following your suggestions, I removed the error translation code. I need some help with the "E2E Tests / Visual Regression" test. Could you either fix it or guide me on how to fix it?

MikesGlitch commented on packages/desktop-client/src/components/settings/Encryption.tsx:
```diff
@@ -79,13 +84,16 @@ export function EncryptionSettings() {
+      primaryAction={<Button isDisabled>{t('Enable encryption…')}</Button>}
+    >
       <Text>
+        <strong>{t('End-to-end encryption')}</strong>{' '}
```
This will be why the E2E tests are failing - You've duplicated the "End-to-end encryption" text. The other text needs translated too

MikesGlitch commented on packages/desktop-client/src/components/settings/Themes.tsx:
```diff
@@ -91,7 +92,8 @@ export function ThemeSettings() {
       <Text>
-        <strong>Themes</strong> change the user interface colors.
+        <strong> {t('Themes')} </strong>
+        {t(' change the user interface colors.')}
```
There's an extra space here - there's already on after _Themes_

MikesGlitch commented on packages/desktop-client/src/components/settings/Encryption.tsx:
```diff
@@ -79,13 +84,16 @@ export function EncryptionSettings() {
+      primaryAction={<Button isDisabled>{t('Enable encryption…')}</Button>}
+    >
       <Text>
+        <strong>End-to-end Encryption is turned on.</strong>{' '}
```
```suggestion
```

MikesGlitch commented on packages/desktop-client/src/components/settings/Encryption.tsx:
```diff
@@ -79,13 +84,16 @@ export function EncryptionSettings() {
+    >
       <Text>
+        <strong>End-to-end Encryption is turned on.</strong>{' '}
         <strong>End-to-end encryption</strong> is not available when running
```
This text needs to be translated

MikesGlitch commented on packages/desktop-client/src/components/settings/Encryption.tsx:
```diff
@@ -79,13 +84,16 @@ export function EncryptionSettings() {
       <Text>
+        <strong>End-to-end Encryption is turned on.</strong>{' '}
         <strong>End-to-end encryption</strong> is not available when running
         without a server. Budget files are always kept unencrypted locally, and
```
This text should be translated

MikesGlitch commented on packages/desktop-client/src/components/settings/Themes.tsx:
```diff
@@ -91,7 +92,8 @@ export function ThemeSettings() {
     >
       <Text>
-        <strong>Themes</strong> change the user interface colors.
+        <strong> {t('Themes')} </strong>
```
```suggestion
        <strong>{t('Themes')} </strong>
```

MikesGlitch commented on packages/desktop-client/src/components/settings/Encryption.tsx:
```diff
@@ -26,11 +27,13 @@ export function EncryptionSettings() {
-          End-to-end Encryption is turned on.
+          {t('End-to-end Encryption is turned on.')}
         </Text>{' '}
         Your data is encrypted with a key that only you have before sending it
```
This still needs translated
![image](https://github.com/user-attachments/assets/5f3d80b9-f963-4125-89a9-27a770236b7f)

awaisalee commented:
@MikesGlitch Could you please guide me on how I checked the image?

<img width="1435" alt="image" src="https://github.com/user-attachments/assets/cc5de1dd-a065-45f1-b014-6204a10426c2">

MikesGlitch commented on packages/desktop-client/src/components/settings/Themes.tsx:
```diff
@@ -91,7 +92,8 @@ export function ThemeSettings() {
       <Text>
-        <strong>Themes</strong> change the user interface colors.
+        <strong>{t('Themes')} </strong>
+        {t('change the user interface colors.')}
```
I think this is the last of it: 

```suggestion
        {t(' change the user interface colors.')}
```

MikesGlitch commented:
> @MikesGlitch Could you please guide me on how I view the image?

Go to the _Summary_ and download the artifact 
![image](https://github.com/user-attachments/assets/37fdb872-8d45-438b-a14c-db5584deeeec)

MikesGlitch commented on packages/desktop-client/src/components/settings/Themes.tsx:
```diff
@@ -91,7 +92,8 @@ export function ThemeSettings() {
     >
       <Text>
-        <strong>Themes</strong> change the user interface colors.
+        <strong>{t('Themes')} </strong>
```
```suggestion
        <strong>{t('Themes')}</strong>
```

MikesGlitch reviewed (approved):
Legend. Thanks for the work!
